packages/compiler-cli/src/ngtsc/typecheck/src/expression.ts

```diff
@@ -96,7 +96,9 @@ class AstTranslator implements AstVisitor {
   visitConditional(ast: Conditional): ts.Expression {
     const condExpr = this.translate(ast.condition);
     const trueExpr = this.translate(ast.trueExp);
-    const falseExpr = this.translate(ast.falseExp);
+    // Wrap `falseExpr` in parens so that the trailing parse span info is not attributed to the
+    // whole conditional.
+    const falseExpr = ts.createParen(this.translate(ast.falseExp));
```

Comment:
Nice catch! Consider using `wrapForDiagnostics` here to make the intent of the parens more clear (beside the comment). It's 

```suggestion
    const falseExpr = wrapForDiagnostics(this.translate(ast.falseExp));
```